package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -537,254 +792,296 @@ def _update_water_selection(self):
 
-        Stores the water bridge data per frame as
-        :attr:`WaterBridgeAnalysis.timeseries` (see there for output
-        format).
 
         Parameters
         ----------
-        start : int (optional)
-            starting frame-index for analysis, ``None`` is the first one, 0.
-            `start` and `stop` are 0-based frame indices and are used to slice
-            the trajectory (if supported) [``None``]
-        stop : int (optional)
-            last trajectory frame for analysis, ``None`` is the last one
-            [``None``]
-        step : int (optional)
-            read every `step` between `start` (included) and `stop` (excluded),
-            ``None`` selects 1. [``None``]
-        verbose : bool (optional)
-             toggle progress meter output
-             :class:`~MDAnalysis.lib.log.ProgressMeter` [``True``]
-        debug : bool (optional)
-             enable detailed logging of debugging information; this can create
-             *very big* log files so it is disabled (``False``) by default;
-             setting `debug` toggles the debug status for
-             :class:`WaterBridgeAnalysis`, namely the value of
-             :attr:`WaterBridgeAnalysis.debug`.
+        atom : groups.Atom
+             heavy atom
 
-        See Also
-        --------
-        :meth:`WaterBridgeAnalysis.generate_table` :
-               processing the data into a different format.
+        Returns
+        -------
+        hydrogen_atoms : AtomGroup or []
+            list of hydrogens (can be a :class:`~MDAnalysis.core.groups.AtomGroup`)
+            or empty list ``[]`` if none were found.
         """
-        self._setup_frames(self.u.trajectory, start, stop, step)
+        try:
+            return atom.residue.atoms.select_atoms(
+                "(name H* 1H* 2H* 3H* or type H) and around {0:f} name {1!s}"
+                "".format(self.r_cov[atom.name[0]], atom.name))
+        except NoDataError:
+            return []
+
+    def logger_debug(self, *args):
+        if self.debug:
+            logger.debug(*args)
+
+
+    def _prepare(self):
+        # The distance for selection is defined as twice the maximum bond length of an O-H bond (2A)
+        # plus order of water bridge times the length of OH bond plus hydrogne bond distance
+        self.selection_distance = 1.0 * (2 * 2 + self.order * (2 + self.distance))
+
+        self._s1_donors = {}
+        self._s1_donors_h = {}
+        self._s1_acceptors = {}
+
+        self._s2_donors = {}
+        self._s2_donors_h = {}
+        self._s2_acceptors = {}
+
+        self.box = self.u.dimensions if self.pbc else None
+        self._update_selection()
+
+        self._water_donors = {}
+        self._water_donors_h = {}
+        self._water_acceptors = {}
+
+        self.timesteps = []
+        if self._s1 and self._s2:
+            self._update_water_selection()
+            logger.info("WBridge analysis: no atoms found in the selection.")
```

Comment:
If the class name is *WaterBridgeAnalysis* then I would also use this name in the logging messages – increases consistency.

Reply:
Changed.